### PR TITLE
[Model] Pixtral Support

### PR DESCRIPTION
**Pull request by AndreSlavescu**
## Summary

This PR aims to support pixtral

## Testing Done

tested model + tested monkey patch

- Hardware Type: 4090
- [ ] run `make test` to ensure correctness
- [x] run `make checkstyle` to ensure code style
- [ ] run `make test-convergence` to ensure convergence

### CODE REVIEW

**Comment by AndreSlavescu** [comment]
Pixtral isn't yet fully supported in transformers library. PR pending release of Pixtral in transformers package.

**Comment by lancerts** [comment]
Exciting! @AndreSlavescu seems it is now supported in the transformer https://github.com/huggingface/transformers/tree/main/src/transformers/models/pixtral,
do you mind update the PR?

**Comment by AndreSlavescu** [comment]
> Exciting! @AndreSlavescu seems it is now supported in the transformer https://github.com/huggingface/transformers/tree/main/src/transformers/models/pixtral, do you mind update the PR?

yes, I'll try to finish this either today or tomorrow

**Comment by icyice69** [comment]
@AndreSlavescu Would it be possible to revisit this? I would be highly interested in using it.

**Review by lancerts — APPROVED** [review]

**Comment by AndreSlavescu** [comment]
@Tcc0403 pinging If you'd like to take a look.

File: src/liger_kernel/transformers/model/pixtral.py
Comment (Tcc0403) [inline src/liger_kernel/transformers/model/pixtral.py:1]: I'm not familiar with pixtral but it looks like it's just a base model. The loss isn't computed in the forward pass, so there's no need to patch CrossEntropy and FusedLinearCrossEntropy.

**Comment by cxmscb** [comment]
Sorry, I wonder when to support the Pixtral.

**Comment by ZhiliangWu** [comment]
Hi @AndreSlavescu , is pixtral still planned to be intergrated?

**Comment by AndreSlavescu** [comment]
> Hi @AndreSlavescu , is pixtral still planned to be intergrated?

I haven't looked into this since the last comment in May, when there was still some core tooling required for the PR to work completely. But I can take a look again.

**Comment by ZhiliangWu** [comment]
> > Hi @AndreSlavescu , is pixtral still planned to be intergrated?
> 
> I haven't looked into this since the last comment in May, when there was still some core tooling required for the PR to work completely. But I can take a look again.

@AndreSlavescu That would be awesome! Looking forward to any coming updates on this.

**Comment by AndreSlavescu** [comment]
@Tcc0403 This is ready for review again

File: test/convergence/fp32/test_mini_models_multimodal.py [hunk 1]
```diff
@@ -1641,3 +1671,160 @@ def test_mini_model_multimodal(
...
+            1e-5,
+            marks=[
+                pytest.mark.skipif(
+                    not PIXTRAL_AVAILABLE, reason="Pixtral not available in this version of transformers"
+                ),
+            ],
+        ),
+    ],
+)
+def test_mini_model_vision(
```
Comment (Tcc0403) [inline test/convergence/fp32/test_mini_models_multimodal.py:1782]: Great idea on seperating vision encoder only test

File: test/convergence/bf16/test_mini_models_with_logits.py
Comment (Tcc0403) [inline test/convergence/bf16/test_mini_models_with_logits.py:1]: No need to change after moving vision encoder only tests under `test_mini_models_multimodal`?

File: test/convergence/fp32/test_mini_models_with_logits.py
Comment (Tcc0403) [inline test/convergence/fp32/test_mini_models_with_logits.py:1]: ditto

File: test/convergence/bf16/test_mini_models_with_logits.py
Comment (AndreSlavescu) [reply test/convergence/bf16/test_mini_models_with_logits.py:1]: ack

Comment (AndreSlavescu) [reply test/convergence/bf16/test_mini_models_with_logits.py:1]: removed changes

File: test/convergence/fp32/test_mini_models_with_logits.py
Comment (AndreSlavescu) [reply test/convergence/fp32/test_mini_models_with_logits.py:1]: removed changes

**Comment by AndreSlavescu** [comment]
> > > Hi @AndreSlavescu , is pixtral still planned to be intergrated?
> > 
> > 
> > I haven't looked into this since the last comment in May, when there was still some core tooling required for the PR to work completely. But I can take a look again.
> 
> @AndreSlavescu That would be awesome! Looking forward to any coming updates on this.
PR merged now
